xayn-ai/src/coi/system.rs

```diff
@@ -214,6 +223,160 @@ impl CoiSystem {
             })
             .collect()
     }
+
+    /// Selects the most relevant key phrases for the coi.
+    ///
+    /// The most relevant key phrases are selected from the set of key phrases of the coi and the
+    /// candidates. The computed relevances are a relative score from the interval `[0, 1]`.
+    #[allow(dead_code)]
+    fn select_key_phrases<
+        CP: CoiPoint + CoiPointKeyPhrases,
+        F: Fn(&str) -> Result<Embedding, Error>,
+    >(
+        &self,
+        coi: &mut CP,
+        candidates: &[String],
```

Comment:
in the way things are currently set up, we can't require this, because `KeyPhrase` holds the smbert embedding of its words (for caching). we could make the smbert available outside of the `CoiSystem`, but then the caller of this function needs to compute the embeddings. this would probably be undesired, not only because it requires to caller to do internal stuff, but also because it may do unnecessary work in computing embeddings for duplicates.

Reply:
Make sense to keep the string outside but then I think this should be one place where we split the function. The user of `CoiSystem` pass the strings but we have one function that create the KeyPhrases from them (with the caching) and then we call the select with the given functions. So we reduce the things that the `select` does.